swifttool/manager.py

```
@@ -0,0 +1,288 @@
+                        # Get the current configuration from rings
+                        ring_devs = self._get_ring_devices(builder, zone)
+                        ip = self._get_zone_ip(devices)
+                        updated_cfg['zones'].setdefault(zone, {})
+                        updated_cfg['zones'][zone].setdefault(ip, {})
+                        updated_cfg['zones'][zone][ip].setdefault('disks', {})
+                        newdevs = []
+                        for dev in devices:
+                            devname = dev['device']
+                            if devname not in ring_devs:
+                                initial_weight = 0
+                            else:
+                                initial_weight = ring_devs[devname]
+                            if initial_weight == dev['target']:
+                                continue
+                            devinfo = {}
+                            if 'metadata' in dev:
+                                devinfo['metadata'] = dev['metadata']
+                            devinfo['blockdev'] = devname
+                            devinfo['weight'] = int(initial_weight +
+                                                    dev['step'])
+                            if abs(devinfo['weight'] - dev['target']) < 50:
+                                devinfo['weight'] = dev['target']
+                            LOG.debug("Setting the %s device capacity to %s",
+                                      devname, str(devinfo['weight']))
+                            newdevs.append(devinfo)
+                            changed = True
+                        updated_cfg['zones'][zone][ip]['disks'][builder] = \
+                            newdevs
+            if changed:
+                ringsdef_helper(updated_cfg, self.metadata, self.outdir)
+            return changed, updated_cfg
+        except Exception as e:
+            print "Error updating cluster configuration", e
```

`print "Error updating cluster configuration" % e`

Also, I don't see catching the exception with a blanket statement like this - what's the advantage?  You're hiding the true cause of the error in this gigantic block of code.  Personally, I see no benefit to hiding this error - I'd bubble it up, not catch it and hide it behind "There was an error!" string.  Also, if there really was an error worth reporting, why print it to stdout?  Print it to stderr.

Gotta be careful with these blanket exception catches.